apps/allocations/app/components/Panel/NewBudget.js

```diff
@@ -1,36 +1,59 @@
 import PropTypes from 'prop-types'
 import React from 'react'
-import { DropDown, TextInput } from '@aragon/ui'
+import { DropDown, IconClose, Info, TextInput, theme } from '@aragon/ui'
 import styled from 'styled-components'
 
 import { Form, FormField } from '../Form'
 import { isStringEmpty } from '../../utils/helpers'
 import { BigNumber } from 'bignumber.js'
-import { MIN_AMOUNT } from '../../utils/constants'
+import { ETH_DECIMALS, MIN_AMOUNT } from '../../utils/constants'
 
 // TODO:: This should be votingTokens from account?
 const INITIAL_STATE = {
   name: '',
   nameError: true,
   amount: '',
   amountError: true,
+  amountOverFunds: false,
   currency: 0,
+  buttonText: 'Create budget'
 }
 
 class NewBudget extends React.Component {
   static propTypes = {
-    onCreateBudget: PropTypes.func.isRequired
+    onCreateBudget: PropTypes.func.isRequired,
+    editingBudget: PropTypes.object,
+    fundsLimit: PropTypes.string.isRequired,
   }
 
-  state =  INITIAL_STATE
+  constructor(props) {
+    super(props)
+    this.state =  INITIAL_STATE
+    if (props.editingBudget) {
+      this.state.name = props.editingBudget.data.name
+      this.state.nameError = false
+      this.state.amount = BigNumber(props.editingBudget.data.amount)
+        .div(ETH_DECIMALS)
+      this.state.amountError = false
+      this.state.currency = props.editingBudget.data.currency === 'ETH' ? 0 : 1 // change this!!
```

Comment:
This is pending fix

Reply:
Yes. Please update this in accordance with the data structure.